climada/hazard/base.py

```diff
@@ -679,6 +680,9 @@ def select(self, event_names=None, date=None, orig=None, reg_id=None, reset_freq
             Select only historical (True) or only synthetic (False) events.
         reg_id : int, optional
             Region identifier of the centroids' region_id attibute.
+        extent: tuple(float, float, float, float), optional
+            Extent of centroids as (min_lon, max_lon, min_lat, max_lat).
```

Comment:
Just a general comment, since I see that it's the same order in the centroids module, but having min_lon, max_lon, min_lat, max_lat order for a bounding box, instead of the classic x_min, y_min, x_max, y_max is a bit confusing, I find.

Reply:
I agree... but this is the code legacy we have.

Reply:
Any strong opinions about it?

Reply:
This always reminds me of the `extent` keyword in matplotlib's `imshow` function which also uses this unusual ordering. I never understood why they decided to do it like this, and why they stick to it.

The cleanest solution would be to require a data class (like `matplotlib.transforms.BboxBase` or `matplotlib.transforms.Bbox`) or at least a named tuple (like `rasterio.coords.BoundingBox`). For backwards compatibility, we could still support a tuple, but raise a deprecation warning, then convert the tuple while assuming the legacy order.

A user doesn't need to import that data class type to use this function, but we could alternatively support a `dict` that is automatically converted to the data class type, as in this example:
```
@dataclass
class BoundingBox:
    left: float
    bottom: float
    right: float
    top: float
```
Usage:
```
>>> extent = dict(left=10, right=20, bottom=0, top=100)
>>> bounds = BoundingBox(**extent)
>>> bounds
BoundingBox(left=10, bottom=0, right=20, top=100)
```
However, we don't need to come to a decision now. Just keep this in mind for future discussions. For now, I would suggest to stick to the confusing legacy tuple format (for backwards compatibility)...

Reply:
Excellent points.